src/batch.js

```diff
@@ -3,26 +3,216 @@
+  if (urlAndQuerystring.length > 1) {
+    const queryStringParams = urlAndQuerystring[1]
+    const keys = queryStringParams.split('&')
+    const queryStrings = keys.map((k) => k.split('='))
+
+    for (let queryString of queryStrings) {
+      if (queryStringParameters[queryString[0]]) {
+        queryStringParameters[queryString[0]].push(queryString[1])
+      } else {
+        queryStringParameters[queryString[0]] = [queryString[1]]
+      }
+    }
+  }
+
+  //Compose event to appear as native AWS event
+  let childEvent = { 
+    ...event,
+    httpMethod: request.method,
+    headers: {...normalizeRequestHeaders(request.headers), authorization: event.headers.authorization},
+    body: request.body,
+    path: request.url,
+    pathParameters: { 
+      proxy: urlAndQuerystring[0].split('/').splice(2).join('/') 
+    },
+    multiValueQueryStringParameters: queryStringParameters,
+    _json: true
+  }
+
+  //Allow consumers to define a custom handler 
+  if (config.handler){
+    return await config.handler(childEvent, {})
+  }
+
+  return await route(childEvent, {})
+}
+
+function normalizeRequestHeaders(reqHeaders = {}) {
+  return Object.keys(reqHeaders).reduce((headers, name) => {
+    headers[name.toLowerCase()] = reqHeaders[name]
+    return headers
+  }, {})
+}
+
+function splitAsyncRequests(orderedRequests) {
+  const asyncRequests = []
+  let syncRequests = []
+  for (let i = 0; i < orderedRequests.length; i++) {
+    if (orderedRequests[i].dependsOn.length > 0) {
+      syncRequests.push(...orderedRequests.splice(i))
+      break
+    }
+
+    if (!orderedRequests.some(req => req.dependsOn.includes(orderedRequests[i].id))) {
+      asyncRequests.push(orderedRequests[i])
+    } else {
+      syncRequests.push(orderedRequests[i])
+    }
+  }
+  return { asyncRequests, syncRequests }
+}
+
+//Depth-first topological sort
+function tSort(requests) {
+  const dependencies = requests.reduce((obj, req) => {
+    obj[req.id] = { dependsOn: [], ...req }
+    return obj
+  }, {})
+  let keys = Object.keys(dependencies)
+  let length
+  let items
+
+  const sorted = new Set()
+  const result = []
+
+  do {
+    length = keys.length
+    items = []
+    // eslint-disable-next-line no-loop-func
+    keys = keys.filter(k => {
+      if (!dependencies[k].dependsOn.every(dependency => sorted.has(dependency))) {
+        return true
+      }
+      items.push(k)
+      return false
+    })
+    result.push(...items)
+    items.forEach(i => sorted.add(i))
+  } while (keys.length && keys.length !== length)
+
+  if (keys.length > 0) {
+    throw new HttpError(400, `invalid body; unable to resolve execution order `+
+      `due to circular dependency chain(s) involving: (${keys.join(', ')})`)
+  }
+
+  return result.map(res => dependencies[res])
 }
 
-function validateBatchRequest(body) {}
+function validateBatchRequest(body, maxBatchSize){
+  // validate json body has been parsed
+  if (typeof body !== 'object' || body === null){
```

Comment:
There is an `HttpError.assert` method that would shorten nearly all of the validation code here.